Modules/System/Math/src/Math.Codeunit.al

```diff
@@ -288,5 +288,82 @@ codeunit 710 Math
     begin
         exit(DotNetMath.Truncate(decimalValue));
     end;
+
+    /// <summary>
+    /// Calculates the bitwise AND value of two integers
+    /// </summary>
+    /// <param name="IntegerValue1">The first integer for the evaluation</param>
+    /// <param name="IntegerValue2">The second integer for the evaluation</param>
+    /// <returns>The resulting integer of the bitwise AND operaion</returns>
+    procedure BitwiseAnd(IntegerValue1: Integer; IntegerValue2: Integer) Result: Integer
+    var
+        BitPosition: Integer;
+        BitIndex: Integer;
+    begin
+        InitBitwiseValues(BitPosition, Result);
+
+        for BitIndex := to 32 do begin
+            if (IntegerValue1 mod 2 = 1) and (IntegerValue2 mod 2 = 1) then
+                UpdateBitwiseResult(BitPosition, Result);
+            UpdateBitwiseValues(IntegerValue1, IntegerValue2, BitPosition);
+        end;
+    end;
+
+    /// <summary>
+    /// Calculates the bitwise OR value of two integers
+    /// </summary>
+    /// <param name="IntegerValue1">The first integer for the evaluation</param>
+    /// <param name="IntegerValue2">The second integer for the evaluation</param>
+    /// <returns>The resulting integer of the bitwise OR operaion</returns>
+    procedure BitwiseOr(IntegerValue1: Integer; IntegerValue2: Integer) Result: Integer
+    var
+        BitPosition: Integer;        
+        BitIndex: Integer;
+    begin
+        InitBitwiseValues(BitPosition, Result);
+
+        for BitIndex := to 32 do begin
+            if (IntegerValue1 mod 2 = 1) or (IntegerValue2 mod 2 = 1) then
+                UpdateBitwiseResult(BitPosition, Result);
+            UpdateBitwiseValues(IntegerValue1, IntegerValue2, BitPosition);
+        end;
+    end;
+
+    /// <summary>
+    /// Calculates the bitwise exclusive OR (XOR) value of two integers
+    /// </summary>
+    /// <param name="IntegerValue1">The first integer for the evaluation</param>
+    /// <param name="IntegerValue2">The second integer for the evaluation</param>
+    /// <returns>The resulting integer of the bitwise exclusive OR (XOR) operaion</returns>
+    procedure BitwiseXOr(IntegerValue1: Integer; IntegerValue2: Integer) Result: Integer
+    var
+        BitPosition: Integer;        
+        BitIndex: Integer;
+    begin
+        InitBitwiseValues(BitPosition, Result);
+        for BitIndex := to 32 do begin
+            if (IntegerValue1 mod 2) + (IntegerValue2 mod 2) = 1 then
+                UpdateBitwiseResult(BitPosition, Result);
+            UpdateBitwiseValues(IntegerValue1, IntegerValue2, BitPosition);
+        end;
+    end;
+
+    local procedure InitBitwiseValues(var BitPosition: Integer; var Result: Integer)
```

Comment:
Create an implementation codeunit and move these local functions there. It wasn't created before because the previous functions either called a dotnet(implementation) or direct value.